src/main/java/org/logstash/dissect/fields/AbstractField.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.dissect.fields;
+
+import com.logstash.Event;
+import org.logstash.dissect.Delimiter;
+import org.logstash.dissect.ValueResolver;
+
+import java.util.Map;
+
+public abstract class AbstractField implements Field {
+    private final int ordinal;
+    protected final String name;
+
+    protected Delimiter join;
+    protected Delimiter next;
```

Comment:
I am on the fence with `protected` members (and even more so if they're mutable). It makes it hard to reason about your code because you have to look into multiple source files to understand who is modifying what at which point.